### PR TITLE
Update to connectrpc.com/connect v1.14.0 and google.golang.org/protobuf v1.32.0

### PR DESCRIPTION
**Pull request by nicksnyder**

### CODE REVIEW

**Comment by nicksnyder** [comment]
@emcfarlane tests are failing, can you take a look?

**Comment by emcfarlane** [comment]
@nicksnyder working. 3 different failure cases that I've corrected, left details in the commits. Will need review.

File: transcoder.go [hunk 1]
```diff
@@ -1249,6 +1249,7 @@ func (w *responseWriter) close() {
 		w.WriteHeader(http.StatusOK)
 	}
 	if w.w != nil {
+		_, _ = w.w.Write(nil) // trigger any final writes
```
[the remark this answers is not in the document]
Comment (emcfarlane) [reply transcoder.go:1252]: Yep sure. The bytes.Buffer `WriteTo` won't `Write` to the `io.Writer` if the payload size is zero: https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/bytes/buffer.go;l=260

Vanguard-go uses the `Write` to trigger the conversion of an empty protomessage (valid zero length payload) to a json message (`{}`). I do think this is an issue with how `vanguard-go` triggers conversions. It shouldn't be dependant on the write mechanism. This was the minimal change I could see to fix the issue but I would presume we wish to revisit later.

File: vanguard_test.go [hunk 2]
```diff
@@ -919,21 +917,16 @@ func runRPCTestCase[Client any](
...
-		if serverInvoked && serverErr != nil {
-			// We expect the server to either have seen the same error or it later
-			// observed a cancel error (since the middleware cancels the request
-			// after it aborts the operation).
-			if connect.CodeOf(serverErr) != connect.CodeOf(expectedErr) && !errors.Is(serverErr, context.Canceled) {
-				assert.Equal(t, connect.CodeCanceled, connect.CodeOf(serverErr))
-			}
-		}
-	default:
+	} else if serverInvoked {
```
[the remark this answers is not in the document]
Comment (emcfarlane) [reply vanguard_test.go:925]: Yep this changed as context error was being set. I thought it was related to https://github.com/connectrpc/connect-go/pull/643 but might have been too hastily in removing it for all cases. Here are the relevant test logs:

```
            --- FAIL: TestTranscoder_BufferTooLargeFails/transforming_writer/must_buffer_response_stream/per_svc_options (0.00s)
                vanguard_test.go:387: WrapStreamingHandler TestTranscoder_BufferTooLargeFails/transforming_writer/must_buffer_response_stream/per_svc_options
                transcoder_test.go:534: RPC error: resource_exhausted: max buffer size (1024) exceeded
                transcoder_test.go:534:
                        Error Trace:    /Users/edward/src/github.com/connectrpc/vanguard-go/vanguard_test.go:932
                                                                /Users/edward/src/github.com/connectrpc/vanguard-go/transcoder_test.go:534
                        Error:          Not equal:
                                        expected: 0x8
                                        actual  : 0x1
                        Test:           TestTranscoder_BufferTooLargeFails/transforming_writer/must_buffer_response_stream/per_svc_options
                transcoder_test.go:534: resource_exhausted: buffer limit exceeded
                transcoder_test.go:534: canceled: context canceled
            --- FAIL: TestTranscoder_BufferTooLargeFails/transforming_writer/must_buffer_response_stream/default_svc_opts (0.00s)
                vanguard_test.go:387: WrapStreamingHandler TestTranscoder_BufferTooLargeFails/transforming_writer/must_buffer_response_stream/default_svc_opts
                transcoder_test.go:534: RPC error: resource_exhausted: max buffer size (1024) exceeded
                transcoder_test.go:534:
                        Error Trace:    /Users/edward/src/github.com/connectrpc/vanguard-go/vanguard_test.go:932
                                                                /Users/edward/src/github.com/connectrpc/vanguard-go/transcoder_test.go:534
                        Error:          Not equal:
                                        expected: 0x8
                                        actual  : 0x1
                        Test:           TestTranscoder_BufferTooLargeFails/transforming_writer/must_buffer_response_stream/default_svc_opts
```

I've pushed up a change to add back `expectServerCancel`. Adding it to the out message passes this test. Does that look correct? It does look like the error should be a resource error though.

**Review by jhump — APPROVED** [review]

**Review by emcfarlane — APPROVED** [review]

**Pull request merged**